manager/integration/tests/test_backing_image.py

```diff
@@ -694,3 +703,199 @@ def test_volume_wait_for_backing_image_condition(client): # NOQA
+        client, BACKING_IMAGE_NAME, bi_url,
+        minNumberOfCopies=1)
+
+    check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 1, "ready")
+
+    backing_image = client.by_id_backing_image(BACKING_IMAGE_NAME)
+    disk_id_1 = next(iter(backing_image.diskFileStatusMap))
+
+    node_1 = get_node_by_disk_id(client, disk_id_1)
+    node_1 = set_node_scheduling_eviction(
+        client, node_1, allowScheduling=False, evictionRequested=True)
+
+    check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 1, "ready")
+    check_backing_image_single_copy_node_eviction(
+        client, BACKING_IMAGE_NAME, node_1)
+
+    backing_image = client.by_id_backing_image(BACKING_IMAGE_NAME)
+    disk_id_2 = next(iter(backing_image.diskFileStatusMap))
+    node_2 = get_node_by_disk_id(client, disk_id_2)
+
+    # Set node1 back to allow scheduling
+    node_1 = set_node_scheduling_eviction(
+        client, node_1, allowScheduling=True, evictionRequested=False)
+
+    # Evict the disk on the node2
+    # and check the backing image file is evicted to node1
+    update_disks = get_update_disks(node_2.disks)
+    for disk in update_disks.values():
+        disk.allowScheduling = False
+        disk.evictionRequested = True
+
+    node_2 = update_node_disks(client, node_2.name, disks=update_disks,
+                               retry=True)
+
+    check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 1, "ready")
+    check_backing_image_single_copy_disk_eviction(
+        client, BACKING_IMAGE_NAME, disk_id_2)
+
+    cleanup_all_backing_images(client)
+
+
+@pytest.mark.backing_image  # NOQA
+def test_backing_image_unable_eviction(client): # NOQA
+    """
+    - Set node1 nodeTag = [node1], diskTag = [node1]
+    - Create a BackingImage with following settings to place the copy on node1
+        - minNumberOfCopies = 1
+        - nodeSelector = [node1]
+        - diskSelector = [node1]
+    - Evict node1
+    - The copy would not be deleted because it is the only copy
+    - The copy can't be copied to other nodes because of the selector settings.
+    """
+    tags_1 = ["node1"]
+    nodes = client.list_node()
+    node_1, node_2, node_3 = nodes
+    set_tags_for_node_and_its_disks(client, node_1, tags_1)
+
+    backing_image_node_selector = ["node1"]
+    backing_image_disk_selector = ["node1"]
+    bi_url = BACKING_IMAGE_RAW_URL
+    create_backing_image_with_matching_url(
+        client, BACKING_IMAGE_NAME, bi_url,
+        minNumberOfCopies=2,
+        nodeSelector=backing_image_node_selector,
+        diskSelector=backing_image_disk_selector)
+    check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 1, "ready")
+
+    backing_image = client.by_id_backing_image(BACKING_IMAGE_NAME)
+    previous_disk_id = next(iter(backing_image.diskFileStatusMap))
+
+    node_1 = set_node_scheduling_eviction(
+        client, node_1, allowScheduling=False, evictionRequested=True)
+
+    time.sleep(60)
```

Comment:
Is there a better way to verify the BI eviction failure? e.g., Checking the related events.

This kind of verification is not deterministic and a waste of test resources.

Reply:
Fixed.